### DISCUSSION TITLE
Enabling customers to create products

### DISCUSSION THREAD

**Discussion by valentin-harrang** [medusajs/medusa#5619] [Q&A]
I am currently engaged in a project utilizing Medusa React, and I'm faced with a unique requirement. Our platform is designed to function as a marketplace, where it's the users (customers) who should be able to offer products, as opposed to administrators.

Currently, the product creation functionality appears to be limited exclusively to admin users. I'm seeking guidance on how to enable this feature for our customer users. Specifically:

- How can I modify or extend user permissions in Medusa to allow non-admin users to create product listings?
- Are there any recommended approaches or best practices for implementing this in a marketplace setting?
- What are the potential security considerations and how can I address them?

Any advice, documentation references, or examples from similar implementations would be greatly appreciated. I'm open to exploring custom development solutions if necessary.

**Comment by srindom** [comment]
We have a Recipe on marketplaces in our documentation which give a high-level path to achieving what you are describing 
 https://docs.medusajs.com/recipes/marketplace
